src/librustc_resolve/late/lifetimes.rs

```diff
@@ -828,6 +831,10 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
             self.insert_lifetime(lifetime_ref, Region::Static);
             return;
         }
+        if self.is_in_const_generic && lifetime_ref.name != LifetimeName::Error {
+            self.emit_non_static_lt_in_const_generic_error(lifetime_ref);
+            return;
```

Comment:
Hmm, I guess we don't have a `RegionKind::Error` to use here... unfortunate. I don't really remember what happens is we fail to store some kind of resolution for a lifetime in type check -- are we relying on not running type check here? Ah, ok, we insert `'static` or an inference variable:

https://github.com/rust-lang/rust/blob/71384101ea3b030b80f7def80a37f67e148518b0/src/librustc_typeck/astconv.rs#L202-L213

I guess that's ok.